src/Database/Type/DateTimeType.php

```diff
@@ -282,15 +282,15 @@ public function manyToPHP(array $values, array $fields, DriverInterface $driver)
             }
 
             $value = $values[$field];
-            if (strpos($value, '0000-00-00') === 0) {
-                $values[$field] = null;
-                continue;
-            }
 
             $class = $this->_className;
             if (is_int($value)) {
                 $instance = new $class('@' . $value);
             } else {
+                if (strpos($value, '0000-00-00') === 0) {
```

Comment:
Due we need this nesting?  Looks like it could be an `elseif` instead.

Reply:
It could be an else if and else.